modules/library/data/src/main/java/org/geotools/data/store/ContentFeatureCollection.java

```diff
@@ -254,6 +318,24 @@ public int size() {
                    count++;
                }
                return count;
+           } else {
+               // we have to iterate, save to cache for later successive readings of data.
+               List<SimpleFeature> tempFeatureCache = new ArrayList<SimpleFeature>();
```

Comment:
I would of made a reader (to cache up to FEATURE_CACHE_LIMIT) and left ContentFeatureCollection out of it. This follow the existing design of using "wrappers".